http-server/nginx-php-web-server/README.md

```diff
@@ -0,0 +1,308 @@
+# Nginx
+
+In this example, we will run a simple Nginx PHP server in a container.
+
+We are installing Nginx on a Ubuntu 16.04 container.
+
+<br>
+
+#### What you need:
+ - Singularity, which you can download and install from [here](https://github.com/sylabs/singularity).
+ - A text editor, like: `micro`, `vim`, or `nano`.
+ - Root access.
+ - A browser to test it, or you can use `w3m`, and install it by `sudo apt-get install w3m w3m-img`.
+
+
+____
+
+<br>
+
+### To start, make the working directory:
+
+```
+$ mkdir ~/nginx
+$ cd ~/nginx/
+```
+
+<br>
+
+Now you can pull the container from the library:
+
+```
+$ singularity pull library://sylabs/examples/nginx.sif:latest
+```
+
+Or you can build from a definition file. Click [here](#building-from-a-definition-file) or scroll down.
+
+<br>
+
+
+### Prepare our directory for the Nginx container:
+
+```
+$ mkdir -p nginx/{lib,log,run,sites-available,tmp,www/html}
+```
+```
+$ touch nginx/{favicon.ico,tmp/data.txt,log/{access.log,error.log},run/nginx.pid}
+```
+
+And prepare for PHP:
+
+```
+$ mkdir -p php/log
+```
+
+```
+$ touch php/{log/php7.0-fpm.log,php7.0-fpm.pid,php7.0-fpm.sock,php.ini}
+```
+
+<br>
+
+### Create the PHP file:
+
+There are two PHP files in this repo.<br>
+We will first test the `index.php` file.
+
+The PHP file will be in `nginx/www/html/index.php`.
+
+```
+$ nano nginx/www/html/index.php
+```
+```
+<!DOCTYPE html>
+<html>
+<head>
+<title>Nginx container test</title>
+</head>
+<body>
+<h1>Hello World</h1>
+<?php
+echo "hello world from PHP!<br>";
+?>
+</body>
+</html>
+```
+
+Or you can download it with `wget`:
+
+```
+$ wget -O nginx/www/html/index.php https://raw.githubusercontent.com/sylabs/examples/master/http-server/nginx-php-web-server/index.php
+```
+
+<br>
+
+### Now, we are almost ready to run it.
+
+There are a lot of bind points, so it's helpful to have a start script.
+
+We will make a start instance script, but you can also shell into the container instead.
+
+```
+$ nano start.sh
+```
+```
+#!/bin/bash
+
+singularity instance start \
+ -B nginx/log/error.log:/var/log/nginx/error.log \
+ -B nginx/log/access.log:/var/log/nginx/access.log \
+ -B nginx/run/nginx.pid:/run/nginx.pid \
+ -B nginx/lib/:/var/lib/nginx/ \
+ -B nginx/favicon.ico:/usr/share/nginx/html/favicon.ico \
+ -B nginx/www/html/index.php:/var/www/html/index.php \
+ -B nginx/tmp/data.txt:/tmp/data.txt \
+ -B php/php.ini:/etc/php/7.0/fpm/php.ini \
+ -B php/:/run/php \
+ -B php/log/php7.0-fpm.log:/var/log/php7.0-fpm.log \
+ nginx.sif nginx php
```

Comment:
Indeed there are many bind points here, I think you could do something similar to what I did for the `mongodb` use case, instead of doing many `-B` tags, set the `SINGULARITY_BINDPATH` with all these in one single command, export it and then in the next one just start the instance. Something like for example: 
```
$ export SINGULARITY_BINDPATH="/from:/to,/data:/mnt"
$ singularity instance start mynginxcontainer.sif nginx php
```
Also, I think the first 2 bind points can be accessed by binding one single folder I mean from `nginx/log` to `/var/log/nginx`